src/store/modules/ModDownloadModule.ts

```diff
@@ -0,0 +1,112 @@
+import {ActionTree, GetterTree} from 'vuex';
+
+import {ImmutableProfile} from '../../model/Profile';
+import ThunderstoreCombo from '../../model/ThunderstoreCombo';
+import ThunderstoreDownloaderProvider from '../../providers/ror2/downloading/ThunderstoreDownloaderProvider';
+import {State as RootState} from '../index';
+import R2Error from "../../model/errors/R2Error";
+
+// TODO: Replace the "enum" in '../../model/enums/StatusEnum' with a proper enum like this one.
+enum StatusEnum {
+    FAILURE = 0,
+    PENDING = 1,
+    SUCCESS = 2,
+}
+
+interface ProgressItem {
+    modName: string;
+    installProgress: number;
+    downloadProgress: number;
+    status: number; // StatusEnum
+    error: R2Error | null;
+}
+
+interface State {
+    downloads: ProgressItem[];
+}
+
+/**
+ * State for Mod Downloading/Installation.
+ */
+export default {
+    namespaced: true,
+
+    state: (): State => ({
+        downloads: [],
+    }),
+
+    getters: <GetterTree<State, RootState>>{
+        activeDownload(state): ProgressItem | undefined {
+            return state.downloads.slice(-1)[0]; // Last element of the array
+        },
+    },
+    mutations: {
+        reset(state: State) {
+            state.downloads = [];
+        },
+        addDownload(state: State, modName: string) {
+            state.downloads.push({
+                modName: modName,
+                installProgress: 0,
+                downloadProgress: 0,
+                status: StatusEnum.PENDING,
+                error: null
+            });
+        },
+        updateDownloadProgress(state: State, params: { progress: number, modName: string, status: number, err: R2Error | null }) {
+            let downloadMod = state.downloads.find((progressItem) => progressItem.modName === params.modName);
```

Comment:
As mentioned in the earlier reviews, .find() returns the first matching item which doesn't work if we have multiple ProgressItems for the same mod (e.g. first one has failed or user installs a different version). To think of it, we therefore probably need some different way of separating the ProgressItems than a simple mod name.